### PR TITLE
AVS 1.0.0 SDK for 2020-03-20 API

### PR DESCRIPTION
**Pull request by ctaggart**
AVS 1.0.0-preview.1 was #12602. This will be 1.0.0 and it updates to the 2020-03-20 stable API:
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/vmware/resource-manager/Microsoft.AVS/stable/2020-03-20/vmware.json

### CODE REVIEW

**Comment by ctaggart** [comment]
I'm headed out on leave for 12 weeks. @leedouglas will be creating a new pull request to replace this one.

**Comment by ctaggart** [comment]
Hello, I'm back from medical leave and need to get this .NET SDK published for AVS, since the service is already GA. I could not figure out how to run autorest from the msbuild targets, so I just ran it directly with:
``` ps1
autorest ../azure-rest-api-specs/specification/vmware/resource-manager/readme.md --csharp --version=v2 --azure-arm=true --use=@microsoft.azure/autorest.csharp@2.3.82 --csharp-sdks-folder=$PWD\sdk --reflect-api-versions
```
Using the latest commit from https://github.com/Azure/azure-rest-api-specs/pull/11317. I'm expecting it to be merged very soon.

My goal in the next couple of days is to add all the tests.

**Comment by ctaggart** [comment]
Hi @markcowl, the additional tests have been added and this is ready for review.

File: sdk/avs/Microsoft.Azure.Management.Avs/src/Microsoft.Azure.Management.Avs.csproj [hunk 1]
```diff
@@ -6,7 +6,7 @@
   <PropertyGroup>
     <PackageId>Microsoft.Azure.Management.Avs</PackageId>
     <Description>Provides developers with libraries for Azure VMware Solution.</Description>
-    <Version>1.0.0-preview.1</Version>
+    <Version>1.0.0-preview.2</Version>
```
Comment (ctaggart) [inline sdk/avs/Microsoft.Azure.Management.Avs/src/Microsoft.Azure.Management.Avs.csproj:9]: need to bump this to 1.0.0

File: sdk/avs/Microsoft.Azure.Management.Avs/src/Microsoft.Azure.Management.Avs.csproj [hunk 2]
```diff
@@ -6,7 +6,7 @@
   <PropertyGroup>
     <PackageId>Microsoft.Azure.Management.Avs</PackageId>
     <Description>Provides developers with libraries for Azure VMware Solution.</Description>
-    <Version>1.0.0-preview.1</Version>
```
Comment (markcowl) [inline sdk/avs/Microsoft.Azure.Management.Avs/src/Microsoft.Azure.Management.Avs.csproj:9]: This is meant to be a non-preview package?  What is the api-version this is based on (should be non-preview).

**Review by markcowl — CHANGES_REQUESTED** [review]
One change, and Missing a few things,
- The generation scripts creates a metadata file in th /eng/mgmt/... folder, please include it
- Please include a link to the swagger PR associated with thsi change in the PR description or comments
- You must add an api version tag for any new api-version in the Az.RP.Sdk.props file for this

File: sdk/avs/Microsoft.Azure.Management.Avs/src/Microsoft.Azure.Management.Avs.csproj (hunk 2 above)
Comment (ctaggart) [reply sdk/avs/Microsoft.Azure.Management.Avs/src/Microsoft.Azure.Management.Avs.csproj:9]: Yes, it is meant to be non-preview. AVS service went GA on 9/28 with this spec: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/vmware/resource-manager/Microsoft.AVS/stable/2020-03-20/vmware.json

**Comment by ctaggart** [comment]
@markcowl, thanks for the review. I made the updates. CI is passing, except for some unrelated issue with the CI infrastructure for MacOS testing.

File: sdk/avs/Microsoft.Azure.Management.Avs/src/Microsoft.Azure.Management.Avs.csproj [hunk 3]
```diff
@@ -6,7 +6,7 @@
...
+    <Version>1.0.0</Version>
```
Comment (markcowl) [inline sdk/avs/Microsoft.Azure.Management.Avs/src/Microsoft.Azure.Management.Avs.csproj:9]: You will need to make a change to at least the AssemblyFileVersion in AssemblyInfo.cs - the file version needs to be monotonically increasing, or some kinds of installation processes (like msis) will not recognize that this assembly is different from the preview.

**Review by markcowl — CHANGES_REQUESTED** [review]

File: sdk/avs/Microsoft.Azure.Management.Avs/src/Microsoft.Azure.Management.Avs.csproj (hunk 3 above)
Comment (ctaggart) [reply sdk/avs/Microsoft.Azure.Management.Avs/src/Microsoft.Azure.Management.Avs.csproj:9]: I set AssemblyFileVersion in 8835f83.

File: sdk/avs/Microsoft.Azure.Management.Avs/src/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,8 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (markcowl) [inline sdk/avs/Microsoft.Azure.Management.Avs/src/Properties/AssemblyInfo.cs:1]: I didn't realize this didn't have an assembly info.  It should have all the appropriate properties in here.  Here is a sample: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/compute/Microsoft.Azure.Management.Compute/src/Properties/AssemblyInfo.cs

Assembly Version should be 1.0.0, Metadata should match this assembly name.

**Review by markcowl — CHANGES_REQUESTED** [review]

File: sdk/avs/Microsoft.Azure.Management.Avs/src/Properties/AssemblyInfo.cs (hunk 4 above)
Comment (ctaggart) [reply sdk/avs/Microsoft.Azure.Management.Avs/src/Properties/AssemblyInfo.cs:1]: I added the additional assembly info metadata as requested in 19b8c13. I set the `AssemblyVersion` to `1.0.0` as well.

**Review by markcowl — APPROVED** [review]

**Comment by ctaggart** [comment]
It has been published:
https://www.nuget.org/packages/Microsoft.Azure.Management.Avs/1.0.0

**Pull request merged**